File: typescript/examples/01-search.ts
/**
 * Example 01 — Case search
 * Language: TypeScript (Node.js)
 * Docs: https://lawdiver.com/docs/api#case-search
 */
import { LawDiverClient } from "../src/client.js";

const client = new LawDiverClient();

const found = await client.search(
  {
    query: "qualified immunity excessive force",
    jurisdiction: { type: "federal_circuit", circuit: "11" },
    limit: 5,
    filters: { dateFrom: "2015-01-01" },
  },
  `search-demo-${crypto.randomUUID()}`, // idempotency key for safe retries
);

console.log(`requestId: ${found.requestId}`);
console.log(`returned ${found.total} of ~${found.totalAvailable ?? "?"} matches`);
console.log(`usage: ${found.usage.quantity} unit(s) · operation=${found.usage.operation}`);
console.log("");

for (const r of found.results) {
  const gl = r.goodLaw
    ? `goodLaw=${r.goodLaw.status}${r.goodLaw.negative ? " (NEGATIVE)" : ""}`
    : "goodLaw=?";
  console.log(`• ${r.caseName}`);
  console.log(`  ${r.citation ?? "(no citation)"} · ${r.courtAbbreviation ?? r.court ?? ""}`);
  console.log(`  ${gl}`);
  if (r.snippet) console.log(`  snippet: ${r.snippet.slice(0, 140)}…`);
  console.log("");
}

if (found.suggestion) {
  console.log("Empty-ish set suggestion:", found.suggestion);
}

File: python/examples/01_search.py
"""Example 01 — Case search (Python).

Docs: https://lawdiver.com/docs/api
"""

from __future__ import annotations

import sys
import uuid
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parents[1]))

from lawdiver import LawDiverClient

with LawDiverClient() as client:
    found = client.search(
        query="qualified immunity excessive force",
        jurisdiction={"type": "federal_circuit", "circuit": "11"},
        limit=5,
        filters={"dateFrom": "2015-01-01"},
        idempotency_key=f"search-demo-{uuid.uuid4()}",
    )

print(f"requestId: {found['requestId']}")
print(f"returned {found['total']} of ~{found.get('totalAvailable', '?')} matches")
usage = found["usage"]
print(f"usage: {usage['quantity']} unit(s) · operation={usage['operation']}")
print()

for r in found["results"]:
    gl = r.get("goodLaw") or {}
    flag = " (NEGATIVE)" if gl.get("negative") else ""
    print(f"• {r.get('caseName')}")
    print(f"  {r.get('citation') or '(no citation)'} · {r.get('courtAbbreviation') or r.get('court') or ''}")
    print(f"  goodLaw={gl.get('status', '?')}{flag}")
    snippet = r.get("snippet") or ""
    if snippet:
        print(f"  snippet: {snippet[:140]}…")
    print()
